enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,470 @@
+  - Select 5 cluster on aws
+
+#### Story 5: The placement API should be scalable to support selecting hundreds of clusters from thousands of clusters.
+
+#### Story 6: A user can select clusters across different regions.
+
+#### Story 7: A controller is able to consume placements to select desired clusters.
+
+### API Spec
+
+#### Placement API
+`Placement` is a namespace scoped API and represents a selection of clusters in one or multiple ManagedClusterSets. When a placement is created, a controller consuming the placement should generate one or multiple placement decisions based on the spec defined in placement.
+```go
+// Placement defines a rule to select a set of ManagedClusters from the ManagedClusterSets bound
+// to the placement namespace.
+//
+// Here is how the placement policy combines with other selection methods to determine a matching
+// list of clusters:
+// 1) Clusters are registered with hub as cluster-scoped ManagedClusters;
+// 2) ManagedClusters are organized into cluster-scoped ManagedClusterSets;
+// 3) ManagedClusterSets are bound to workload namespaces;
+// 4) Namespace-scoped Placements specify a slice of ManagedClusterSets which select a working set
+//    of potential ManagedClusters;
+// 5) Then Placements subselect from that working set using name/label/claim selection.
+//
+// No ManagedClusters will be selected if there is no ManagedClusterSet bound to the placement
+// namespace. User is able to bind a ManagedClusterSet to a namespace in two ways:
+// 1) Create a ManagedClusterSetBinding in the namespace if they have an RBAC rule to CREATE
+//    on the virtual subresource of `managedclustersets/bind`;
+// 2) Create a service account with name 'placement', and create a ClusterRoleBinding to grant
+//    the CREATE permission on the virtual subresource `managedclustersets/bind` of a ManagedClusterSet
+//    to this service account;
+//
+// A slice of PlacementDecisions with label placement.open-cluster-management.io={placement name}
+// will be created to represent the cluster selected by this placement.
+type Placement struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of Placement.
+	Spec PlacementSpec `json:"spec"`
+
+	// Status represents the current status of the Placement
+	// +optional
+	Status PlacementStatus `json:"status,omitempty"`
+}
+
+// PlacementSpec defines the attributes of Placement.
+// An empty PlacementSpec selects all clusters from the bound ManagedClusterSets to placement namespace.
+type PlacementSpec struct {
+	// Predicates represent a slice of predicates to select clusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+
+	// Affinity represents the scheduling constraints
+	// +optional
+	Affinity *Affinity `json:"affinity,omitempty"`
+}
+
+// ClusterPredicate represents a predicate to select clusters.
+type ClusterPredicate struct {
```

Comment:
I am interested in adding a time window to the predicate.
I'm not ready to say we DON'T show the placement target, but at least in the status, mark the predicate list or grouping as active/inactive(blocked)

I could also be convinced this is a top level object, and just applies to the entire placement rule.

https://github.com/open-cluster-management/multicloud-operators-subscription/blob/84dfe430767e453872ccdb19218b3ccb1ab8e913/deploy/crds/apps.open-cluster-management.io_subscriptions.yaml#L321

Reply:
I think it is an interesting use case. But I would limit the scope of this proposal as-is for now, and treated this case as a following proposal. WDYT?

@elgnay do you think we could put it as a use case in this doc and set it as API beta graduation criteria.

Reply:
I've added a user story for time window support and set it as API beta graduation criteria. We may need an additional enhancement to cover it.